data/items.js

```diff
@@ -2376,7 +2376,9 @@ exports.BattleItems = {
 		id: "mail",
 		name: "Mail",
 		spritenum: 403,
-		onTakeItem: false,
+		onTakeItem: function(item, source) {
+			if (!source.volatiles['knockoff']) return false;
```

Comment:
It's probably better to check the battle's `activeMove` instead of pinning a volatile on Knock Off just for the purposes of Mail.